### ISSUE TITLE
Inconsistent handling of `.data` by `object_usage_linter`

### ISSUE THREAD

**Issue by pbbgss** [jimhester/lintr#3045]
The `.data` pronoun is flagged as an object without a visible binding when used in the definition of a named function but not when used in the definition of an anonymous function or outside of a function.

``` r
# No lint
lintr::lint(
  text = 'iris |> dplyr::filter(.data$Species == "setosa")',
  linters = lintr::object_usage_linter()
)
#> ℹ No lints found.

# No lint
lintr::lint(
  text = 'function() {iris |> dplyr::filter(.data$Species == "setosa")}',
  linters = lintr::object_usage_linter()
)
#> ℹ No lints found.

# Lint!
lintr::lint(
  text = 'fun <- function() {iris |> dplyr::filter(.data$Species == "setosa")}',
  linters = lintr::object_usage_linter()
)
#> <text>:1:42: warning: [object_usage_linter] no visible binding for global variable '.data'
#> fun <- function() {iris |> dplyr::filter(.data$Species == "setosa")}
#>                                          ^~~~~
```

<sup>Created on 2026-05-18 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

**Comment by MichaelChirico** [comment]
we only search assigned functions.

if this is a package, importFrom(rlang, .data) should suffice for the false positives.

**Comment by pbbgss** [comment]
Thanks for the speedy response - it is not in a package, sadly. Any appetite to add "exceptions" to the object usage linter? For now I think my best bet is to skip that linter on function definitions where `.data` is used...

**Comment by MichaelChirico** [comment]
In a script, adding `library(rlang)`, `.data = NULL`, or `.data = rlang::.data` would all work, IINM.
